friendsController.js

```diff
@@ -0,0 +1,42 @@
+const friends = require('./friends');
+const router = require('express').Router();
+
+
+
+router.get('/',(req,res) => {
+    friends.find({},(err,results) => {
+        res.status(200).send(results)
+    })
+})
+
+router.post('/',(req,res)=> {
```

Comment:
Per the POST endpoint's specs, we need to account for missing fields in our `req.body`. Also, need error handling, and since you're using a callback function instead of Promises (`then`/`catch`), we have to handle Mongoose's Error object (http://mongoosejs.com/docs/api.html#Error) in this manner:

```
router.post('/', (req, res) => {
  const { age, firstName, lastName } = req.body
  if (!age || !firstName || !lastName) {
    return res.status(400).json({ errorMessage: "Please provide firstName, lastName and age for the friend." })
  }

  friends.create(req.body, (err, result) => {
    if (err && err.name === 'MongoNetworkError') {
      res.status(500).json({ errorMessage: "There was an error while saving the friend to the database." })
    } else if (err && err.name === 'ValidationError') {
      res.status(400).json({ errorMessage: "Age must be a number between 1 and 120" })
    } else {
      res.status(201).json(result)
    }
  })
})
```